Makefile

```diff
@@ -0,0 +1,89 @@
+################################################################################
+##                             VERSION PARAMS                                 ##
+################################################################################
+
+## Docker Build Versions
+DOCKER_BUILDER_SERVER_IMAGE = golang:1.14.4
+DOCKER_BASE_IMAGE = alpine:3.12
+
+################################################################################
+
+export GOBIN ?= $(PWD)/bin
+GO ?= $(shell command -v go 2> /dev/null)
+GOFLAGS ?= $(GOFLAGS:)
+PILLAR_IMAGE ?= mattermost/pillar:test
+
+# Build
+MACHINE = $(shell uname -m)
+BUILD_TIME := $(shell date -u +%Y%m%d.%H%M%S)
+BUILD_HASH = $(shell git rev-parse HEAD)
+
+# Tests
+TESTS=.
+
+export GO111MODULE=on
+
+## Checks the code style, tests, builds and bundles.
+all: check-style
+
+## Build the docker image for pillar
+.PHONY: build-image
+build-image:
+	@echo Building Pillar Container Image
+	docker build \
+	--build-arg DOCKER_BUILDER_SERVER_IMAGE=$(DOCKER_BUILDER_SERVER_IMAGE) \
+	--build-arg DOCKER_BASE_IMAGE=$(DOCKER_BASE_IMAGE) \
+	--build-arg ENV=cloud \
+	--build-arg GITHUB_USERNAME=$(GITHUB_USERNAME) \
+	--build-arg GITHUB_TOKEN=$(GITHUB_TOKEN) \
+	. -t $(PILLAR_IMAGE) \
+	--no-cache
+
+## Build the Container image for local development.
+.PHONY: dev-start
+dev-start:
+	@echo Starting local development
+	docker-compose up -d
+
+## Shutdown the development environment.
+.PHONY: dev-stop
+dev-stop:
+	@echo Shutting down the local environment
+	docker-compose down
+
+## Clean the development environment.
+.PHONY: dev-clean
+dev-clean:
+	@echo Cleaning the local environment
+	docker-compose kill
```

Comment:
Do we plan to add `docker-compose` in the future? Otherwise I think we should remove those rules to avoid confusion.